### PR TITLE
more logging improvements

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Docs PR: https://github.com/arangodb/docs/pull/617
Enterprise companion PR: https://github.com/arangodb/enterprise/pull/646

More improvements for logging:
  
* Added new REST API endpoint GET `/_admin/log/entries` to return log entries in a more intuitve format, putting each log entry with all its properties into an object. The API response is an array with all log message objects that match the search criteria.
  This is an extension to the already existing API endpoint GET `/_admin/log`, which returned log messages fragmented into 5 separate arrays.

 The already existing API endpoint GET `/_admin/log` for retrieving log messages is now deprecated, altough it will stay available for some time.

* Truncation of log messages now takes JSON format into account, so that the truncation of oversized JSON log messages still keeps a valid JSON structure even after the truncation.

* The maximum size of in-memory log messages was doubled from 256 to 512 chars, so that longer parts of each log message can be preserved now.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required

#### Related Information

- [x] Docs PR: https://github.com/arangodb/docs/pull/617
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/646

### Testing & Verification

- [x] The behavior in this PR was *manually tested*
- [x] This PR adds tests that were used to verify all changes:
  - [x] Added new **integration tests** (e.g. in shell_client, server_parameters)

Link to Jenkins PR:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13878/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13881/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13885/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13887/

**Comment by jsteemann** [comment]
Tests blue

File: Documentation/DocuBlocks/Rest/Administration/get_admin_log.md [hunk 1]
```diff
@@ -40,7 +103,12 @@ imposes a chronological order. The default value is *asc*.
 
 @RESTDESCRIPTION
 Returns fatal, error, warning or info log messages from the server's global log.
-The result is a JSON object with the following attributes:
+The result is a JSON object with the attributes described below.
+
+This API can be turned off via the startup option `--log.api-enabled`. In case
+the API is disabled, all requests will be responded to with HTTP 403. If the
+API is enabled, accessing it requires admin privileges, or even superuser
+privileges, depending on the value of the `--log.api-enabled` startup option.
```
Comment (jsteemann) [inline Documentation/DocuBlocks/Rest/Administration/get_admin_log.md:112]:
```suggestion
privileges, depending on the value of the `--log.api-enabled` startup option.

This API is **deprecated** in favor of the `/_admin/log/entries` REST API, which 
provides the same data in a more intuitive and easier to process format.

```

**Review by fceller — APPROVED** [review]
LGTM

**Pull request merged**